### PR TITLE
Fix broken links dependency for prometheus component

### PR DESCRIPTION
**Pull request by chotiwat**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

I've encountered the below error when prometheus is enabled but kube-scheduler is disabled. Specifically, I was trying to enable prometheus for `kube-scheduler-simulator` so this issue is reproducible by setting `options.prometheusPort` to `9090` in https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/6cce7f41e06d3a419188f689264387d937c8b460/kwok.yaml.

```
{"time":"2024-09-17T01:40:26.594777384Z","level":"ERROR","source":{"function":"main.main","file":"/workspace/cmd/kwokctl/main.go","line":54},"msg":"Execute exit","err":"failed to start cluster \"kwok-kwok\": broken links dependency detected: [prometheus]"}
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wzshiming** [comment]
Please run `./hack/update-testdata.sh` to update the test data

File: test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt [hunk 1]
```diff
@@ -6,16 +6,16 @@ mkdir -p <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/pki
...
 mkdir -p <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/etcd
 # Download https://github.com/etcd-io/etcd/releases/download/v3.5.15/etcd-v3.5.15-<OS>-<ARCH>.<TAR> and extract etcd to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/etcd
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-apiserver to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-apiserver
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-controller-manager to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-controller-manager
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-scheduler to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-scheduler
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-apiserver to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-apiserver
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-controller-manager to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-controller-manager
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-scheduler to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-scheduler
```
Comment (wzshiming) [inline test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt:11]: Revert this

File: test/e2e/kwokctl/dryrun/testdata/binary/create_cluster_with_extra.txt [hunk 2]
```diff
@@ -6,9 +6,9 @@ mkdir -p <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/pki
...
 mkdir -p <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/etcd
 # Download https://github.com/etcd-io/etcd/releases/download/v3.5.15/etcd-v3.5.15-<OS>-<ARCH>.<TAR> and extract etcd to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/etcd
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-apiserver to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-apiserver
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-controller-manager to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-controller-manager
-# Download https://dl.k8s.io/release/v1.31.0/bin/<OS>/<ARCH>/kube-scheduler to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-scheduler
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-apiserver to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-apiserver
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-controller-manager to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-controller-manager
+# Download https://github.com/kwok-ci/k8s/releases/download/v1.31.0-kwok.0-<OS>-<ARCH>/kube-scheduler to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kube-scheduler
```
Comment (wzshiming) [inline test/e2e/kwokctl/dryrun/testdata/binary/create_cluster_with_extra.txt:11]: Revert this

File: test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt [hunk 3]
```diff
@@ -6,16 +6,16 @@ mkdir -p <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/pki
...
 # Download https://github.com/kubernetes-sigs/kwok/releases/download/v0.7.0/kwok-<OS>-<ARCH> to <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/bin/kwok-controller
 cat <<EOF ><ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/kubeconfig.yaml
 apiVersion: v1
 clusters:
 - cluster:
     certificate-authority: <ROOT_DIR>/workdir/clusters/<CLUSTER_NAME>/pki/ca.crt
-    server: https://127.0.0.1:32764
+    server: https://127.0.0.1:32763
```
Comment (wzshiming) [inline test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt:18]: Please delete the existing kwok cluster before updating testdata

File: test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt (hunk 1 above)
Comment (chotiwat) [reply test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt]: Why did this happen, even on the clean `main` branch?

Comment (wzshiming) [reply test/e2e/kwokctl/dryrun/testdata/binary/create_cluster.txt]: https://kwok.sigs.k8s.io/docs/user/kwokctl-platform-specific-binaries/

k8s doesn't officially provide Darwin binaries, the binary runtime under Darwin uses unofficial builds on https://github.com/kwok-ci/k8s by default.

**Comment by wzshiming** [comment]
please squash commits

**Pull request merged**